types/node/globals.d.ts

```diff
@@ -664,7 +664,7 @@ declare namespace NodeJS {
     type ExitListener = (code: number) => void;
     type RejectionHandledListener = (promise: Promise<any>) => void;
     type UncaughtExceptionListener = (error: Error) => void;
-    type UnhandledRejectionListener = (reason: {} | null | undefined, promise: Promise<any>) => void;
+    type UnhandledRejectionListener = (reason: Error | any, promise: Promise<any>) => void;
```

Comment:
This is actually the same as `any`.
The reason why `any` was remove here is to enforce users to do type checking before they access any property.
Do you have a specific use case where the current definition is not working?

Reply:
Hi, thanks for comment. 
{} | null | undefined  is **not** actually the same as any. because {} means type does not have any properties. 
Yes, i have case where this not work properly.
```typescript
process.on('unhandledRejection', (reason: any) => {
  let text = reason;
  let extra: {} = {};
  if (reason.message) {
    const { message, ...rest } = reason;
    text = message;
    extra = rest;
  }
  logger.error(`### Unhandled rejection: ${text}`, extra);
});
``` 

`const { message, ...rest } = reason;` i can not destruct message property from reason due to `{} | null | undefined`

Reply:
@Flarna 
you can also take a look on Node manual 
https://nodejs.org/api/process.html#process_event_unhandledrejection

Reply:
I know that `{} | null | undefined` is not the same as `any`. But this is by intention.
Your code above will most likely crash if someone does `throw undefined` or `throw null`.
In short you should always do a type check first and then cast. This is what `unknown` is good for.

Reply:
The node manual is not using the typescript type system therefore you wont find `unknown` there.

Reply:
do you mean something like this 
```typescript 
  if (reason?.message) {
    const { message, ...rest } = reason;
    text = message;
    extra = rest;
  }
```